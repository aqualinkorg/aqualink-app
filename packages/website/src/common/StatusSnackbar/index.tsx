import React from 'react';
import {
  Button,
  Snackbar,
  Theme,
  Alert,
  SnackbarCloseReason,
  AlertProps,
} from '@mui/material';
import makeStyles from '@mui/styles/makeStyles';

const StatusSnackbar = ({
  open,
  message,
  furtherActionLabel,
  severity,
  handleClose,
  onFurtherActionTake = () => {},
}: StatusSnackbarProps) => {
  const classes = useStyles({ hasMessage: !!message });

  return message ? (
    <Snackbar
      className={classes.snackbar}
      open={open}
      onClose={handleClose}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
    >
      <Alert
        className={classes.alert}
        variant="filled"
        onClose={(e) => handleClose(e)}
        severity={severity}
        classes={{ message: classes.alertMessage }}
      >
        {message}
        {furtherActionLabel && onFurtherActionTake && (
          <Button
            size="small"
            className={classes.button}
            onClick={onFurtherActionTake}
          >
            {furtherActionLabel}
          </Button>
        )}
      </Alert>
    </Snackbar>
  ) : null;
};

const useStyles = makeStyles<Theme, { hasMessage: boolean }>(
  (theme: Theme) => ({
    snackbar: {
      maxWidth: '50%',
      [theme.breakpoints.down('md')]: {
        maxWidth: '90%',
      },
    },
    alert: {
      alignItems: 'center',
    },
    alertMessage: ({ hasMessage }) => ({
      display: 'flex',
      alignItems: 'center',
      ...(hasMessage ? { padding: 0 } : {}),
    }),
    button: {
      marginLeft: theme.spacing(1.5),
    },
  }),
);

interface StatusSnackbarProps {
  open: boolean;
  message?: string;
  furtherActionLabel?: string;
  severity: AlertProps['severity'];
  handleClose: (
    event: React.SyntheticEvent<any> | Event,
    reason?: SnackbarCloseReason,
  ) => void;
  onFurtherActionTake?: () => void;
}

export default StatusSnackbar;
