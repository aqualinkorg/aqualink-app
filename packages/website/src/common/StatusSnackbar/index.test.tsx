import React from "react";
import { render } from "@testing-library/react";

import StatusSnackbar from ".";

test("renders as expected", () => {
  const { container } = render(
    <StatusSnackbar
      message="All good!!"
      severity="success"
      handleClose={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});
