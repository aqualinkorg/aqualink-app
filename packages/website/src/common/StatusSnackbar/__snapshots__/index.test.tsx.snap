// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-snackbar
    anchororigin="[object Object]"
    classname="makeStyles-snackbar-1"
    open="true"
  >
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-filledSuccess makeStyles-alert-2 MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message makeStyles-alertMessage-3 makeStyles-alertMessage-5"
      >
        All good!!
      </div>
      <div
        class="MuiAlert-action"
      >
        <mock-iconbutton
          aria-label="Close"
          color="inherit"
          size="small"
          title="Close"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </mock-iconbutton>
      </div>
    </div>
  </mock-snackbar>
</div>
`;
